core/base/src/TSystem.cxx

```diff
@@ -1102,6 +1103,7 @@ const char *TSystem::ExpandFileName(const char *fname)
 
 Bool_t TSystem::ExpandFileName(TString &fname)
 {
+   R__LOCKGUARD(gGlobalMutex); // NEWLY_ADDED
```

Comment:
That one should be needed either.  If it is somehow helpful, then this would indicates that we need a micro-lock somewhere deeper inside `TSystem::ExpandFileName`